pkg/k8sutil/create.go

```diff
@@ -161,3 +168,60 @@ func parseJSONManifest(data []byte) ([]manifest, error) {
 	}
 	return manifests, nil
 }
+
+// CreateOrUpdateNamespace creates the release namespace or updates the namespace
+// if it already exists.
+func CreateOrUpdateNamespace(ns Namespace, nsclient corev1typed.NamespaceInterface) error {
+	if ns.Name == "" {
+		return fmt.Errorf("namespace name can't be empty")
+	}
+
+	// Ensure the namespace in which we create release and resources exists.
+	_, err := nsclient.Create(context.TODO(), &v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        ns.Name,
+			Labels:      ns.Labels,
+			Annotations: ns.Annotations,
+		},
+	}, metav1.CreateOptions{})
+	if err != nil && !apierrors.IsAlreadyExists(err) {
+		return err
```

Comment:
Please decorate errors to make tracing easier.
```suggestion
		return fmt.Errorf("creating namespace %q: %w", ns.Name, err)
```